Build failure: pkgsLLVM.libomxil-bellagio

### Steps To Reproduce

Steps to reproduce the behavior:
1. build `pkgsLLVM.libomxil-bellagio`

### Build log

```
/nix/store/bpvgbcdp4a3m200m69y59wbby19gncwy-bash-5.2p26/bin/bash ../../libtool  --tag=CC   --mode=compile aarch64-unknown-linux-gnu-clang -DHAVE_CONFIG_H -I. -I../..    -I../../include -I../../src -Wall -Werror -DCONFIG_DEBUG_LEVEL=0 -c -o libomxbase_la-omx_base_filter.lo `test -f 'omx_base_filter.c' || echo './'`omx_base_filter.c
libtool: compile:  aarch64-unknown-linux-gnu-clang -DHAVE_CONFIG_H -I. -I../.. -I../../include -I../../src -Wall -Werror -DCONFIG_DEBUG_LEVEL=0 -c omx_base_filter.c  -fPIC -DPIC -o .libs/libomxbase_la-omx_base_filter.o
/nix/store/bpvgbcdp4a3m200m69y59wbby19gncwy-bash-5.2p26/bin/bash ../../libtool  --tag=CC   --mode=compile aarch64-unknown-linux-gnu-clang -DHAVE_CONFIG_H -I. -I../..    -I../../include -I../../src -Wall -Werror -DCONFIG_DEBUG_LEVEL=0 -c -o libomxbase_la-omx_base_sink.lo `test -f 'omx_base_sink.c' || echo './'`omx_base_sink.c
libtool: compile:  aarch64-unknown-linux-gnu-clang -DHAVE_CONFIG_H -I. -I../.. -I../../include -I../../src -Wall -Werror -DCONFIG_DEBUG_LEVEL=0 -c omx_base_sink.c  -fPIC -DPIC -o .libs/libomxbase_la-omx_base_sink.o
omx_base_sink.c:77:35: error: variable 'inBufExchanged' set but not used [-Werror,-Wunused-but-set-variable]
   77 |   int                             inBufExchanged              = 0;
      |                                   ^
1 error generated.
make[3]: *** [Makefile:331: libomxbase_la-omx_base_sink.lo] Error 1
make[3]: Leaving directory '/build/libomxil-bellagio-0.9.3/src/base'
make[2]: *** [Makefile:627: all-recursive] Error 1
make[2]: Leaving directory '/build/libomxil-bellagio-0.9.3/src'
make[1]: *** [Makefile:381: all-recursive] Error 1
make[1]: Leaving directory '/build/libomxil-bellagio-0.9.3'
make: *** [Makefile:266: all] Error 2
```

### Additional context

Found while compiling `pkgsLLVM.mesa` in nixpkgs staging.

### Notify maintainers

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

@vcunat 

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
 - system: `"aarch64-linux"`
 - host os: `Linux 5.15.156, NixOS, 24.05 (Uakari), 24.05.20240418.bc279bb`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.2`
 - nixpkgs: `/run/current-system/nixpkgs`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/libraries/libomxil-bellagio/default.nix
+++ b/pkgs/development/libraries/libomxil-bellagio/default.nix
@@ -35,7 +35,9 @@ stdenv.mkDerivation rec {
   env.NIX_CFLAGS_COMPILE =
     # stringop-truncation: see https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1028978
     if stdenv.cc.isGNU then "-Wno-error=array-bounds -Wno-error=stringop-overflow=8 -Wno-error=stringop-truncation"
-    else "-Wno-error=absolute-value -Wno-error=enum-conversion -Wno-error=logical-not-parentheses -Wno-error=non-literal-null-conversion";
+    else let
+      isLLVM17 = stdenv.cc.bintools.isLLVM && lib.versionAtLeast stdenv.cc.bintools.version "17";
+    in "-Wno-error=absolute-value -Wno-error=enum-conversion -Wno-error=logical-not-parentheses -Wno-error=non-literal-null-conversion${lib.optionalString (isLLVM17) " -Wno-error=unused-but-set-variable"}";
 
   meta = with lib; {
     homepage = "https://omxil.sourceforge.net/";

